datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ResolverUtils.java

```diff
@@ -81,20 +83,73 @@ public static Map<String, String> buildFacetFilters(@Nullable List<FacetFilterIn
             if (!validFacetFields.contains(facetFilterInput.getField())) {
                 throw new ValidationException(String.format("Unrecognized facet with name %s provided", facetFilterInput.getField()));
             }
-            facetFilters.put(facetFilterInput.getField(), facetFilterInput.getValue());
+            facetFilters.put(facetFilterInput.getField(), facetFilterInput.getValues().get(0));
         });
 
         return facetFilters;
     }
 
+    // In the case that user sends filters to be or-d together, we need to build a series of conjective criterion
+    // arrays, rather than just one for the AND case.
+    public static ConjunctiveCriterionArray buildConjunctiveCriterionArrayWithOr(
+        final List<Criterion> andCriterions,
+        @Nonnull List<FacetFilterInput> orFilters
+    ) {
+        return new ConjunctiveCriterionArray(orFilters.stream().map(orFilter -> {
+                CriterionArray andCriterionsForOr = new CriterionArray(andCriterions);
+                andCriterionsForOr.add(criterionFromFilter(orFilter));
+
+                return new ConjunctiveCriterion().setAnd(
+                    andCriterionsForOr
```

Comment:
So here we are doing something like this...

If you pass a list of and criterion:

`[ field equals value, field2 equals value2 ]`

and some set of ORs

`[ field3 equals value3 ]`

then we will essentially build an OR of ANDs?

`OR [  [ field equals value] AND [ field3 equals value3 ],  [field2 equals value2 ] AND [ field3 equals value3 ]  ]`

Reply:
So I had to work through this in my head for some time. We should consider adding some comments or something to give an example !

Reply:
resovled!